src/main/java/com/eprosima/fastdds/idl/templates/DDSPubSubTypeHeader.stg

```diff
@@ -262,11 +262,11 @@ private:
         return $struct.maxXCDRv2PlainTypeSerializedSize$ULL ==
                (detail::$struct.name$_offset_of<$struct.name$, detail::$struct.name$_f>() +
                sizeof($last(struct.members).typecode.cppTypename$));
-        $elseif(struct.inheritances)$
-        $if(last(struct.inheritances).isPlain)$
+        $elseif(struct.inheritance)$
+        $if(struct.inheritance.isPlain)$
```

Comment:
Join these conditional in only one. ST4 supports it.

Reply:
See my previous response.